test/e2e/reconciler/apiexportendpointslice/apiexportendpointslice_test.go

```diff
@@ -0,0 +1,239 @@
+		},
+		Spec: apisv1alpha1.APIExportSpec{},
+	}
+
+	slice := &apisv1alpha1.APIExportEndpointSlice{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "my-slice",
+		},
+		Spec: apisv1alpha1.APIExportEndpointSliceSpec{
+			APIExport: apisv1alpha1.ExportBindingReference{
+				Path: e.exportClusterPath.String(),
+				Name: export.Name,
+			},
+		},
+	}
+
+	partition := &topologyv1alpha1.Partition{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "my-partition",
+		},
+		Spec: topologyv1alpha1.PartitionSpec{
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"region": "apiexportendpointslice-test-region",
+				},
+			},
+		},
+	}
+
+	t.Logf("Creating an APIExportEndpointSlice with reference to a nonexistent APIExport")
+	sliceClient := e.kcpClusterClient.ApisV1alpha1().APIExportEndpointSlices()
+
+	_, err := sliceClient.Cluster(e.partitionClusterPath).Create(ctx, slice, metav1.CreateOptions{})
+	require.Error(t, err, "no error creating APIExportEndpointSlice (admission should have declined it)")
+
+	t.Logf("Creating the missing APIExport")
+	exportClient := e.kcpClusterClient.ApisV1alpha1().APIExports()
+	_, err = exportClient.Cluster(e.exportClusterPath).Create(ctx, export, metav1.CreateOptions{})
+	require.NoError(t, err, "no error creating APIExport")
+
+	t.Logf("Retrying to create the APIExportEndpointSlice after the APIExport has been created")
+	require.Eventually(t, func() bool {
+		slice, err = sliceClient.Cluster(e.partitionClusterPath).Create(ctx, slice, metav1.CreateOptions{})
+		return err == nil
+	}, wait.ForeverTestTimeout, 100*time.Millisecond, "expected APIExportEndpointSlice creation to succeed")
+
+	require.Eventually(t, func() bool {
+		s, err := e.kcpClusterClient.Cluster(e.partitionClusterPath).ApisV1alpha1().APIExportEndpointSlices().Get(ctx, slice.Name, metav1.GetOptions{})
+		require.NoError(t, err)
+		return conditions.IsTrue(s, apisv1alpha1.APIExportValid)
+	}, wait.ForeverTestTimeout, 100*time.Millisecond, "expected valid APIExport")
+
+	t.Logf("Adding a Partition to the APIExportEndpointSlice")
+	patchedSlice := slice.DeepCopy()
+	patchedSlice.Spec.Partition = partition.Name
+	mergePatch, err := jsonpatch.CreateMergePatch(encodeJSON(t, slice), encodeJSON(t, patchedSlice))
+	require.NoError(t, err)
+
+	t.Logf("Try to patch the APIExportEndpointSlice to point at a Partition")
+	slice, err = e.kcpClusterClient.Cluster(e.partitionClusterPath).ApisV1alpha1().APIExportEndpointSlices().Patch(ctx, slice.Name, types.MergePatchType, mergePatch, metav1.PatchOptions{})
+	require.NoError(t, err, "error patching APIExportEndpointSlice")
+
+	require.Eventually(t, func() bool {
+		s, err := e.kcpClusterClient.Cluster(e.partitionClusterPath).ApisV1alpha1().APIExportEndpointSlices().Get(ctx, slice.Name, metav1.GetOptions{})
+		require.NoError(t, err)
+		return conditions.IsFalse(s, apisv1alpha1.PartitionValid) && conditions.GetReason(s, apisv1alpha1.PartitionValid) == apisv1alpha1.PartitionInvalidReferenceReason
```

Comment:
shouldn't we examine APIExportEndpoints with invalid partition?

Reply:
done